### PR TITLE
Refactor/validate login register form

### PR DESCRIPTION
**Pull request by AdrieneTZ**
- Move validating request body data from `user-controller.js` to `middlewares/validate-req.js`
- The previous validation processes are in `user-controller.js/login` and `user-controller.js/register`, and that means requests are passing `passport.js` middleware, then going to the user controller.
- Now the validation processes are: `validate request body data` > `passport` > `user controller`

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -3,21 +3,37 @@ const router = express.Router()
...
+const {
+  validateRegisterRequestBody,
+  validateLoginRequestBody,
+} = require('../middlewares/validate-req')
 
 // User login
-router.post('/login', passport.authenticate('local', { session: false }), userController.login)
+router.post(
+  '/login',
+  validateLoginRequestBody,
```
Comment (seanlin1125) [inline routes/index.js:26]: 這樣應該沒問題，等於確認格式後再進行Passport帳密驗證

File: controllers/user-controller.js [hunk 2]
```diff
@@ -67,128 +24,53 @@ const userController = {
...
-            isDeleted: false,
+            approvalStatus: 'reviewing',
+            permissionRole: 'user',
           },
         })
+
         delete user.password
-        return res.status(201).json(user)
+        return res.status(201).json({
+          status: 'Success',
```
Comment (seanlin1125) [inline controllers/user-controller.js:36]: success小寫就好?統一一下?

File: controllers/user-controller.js [hunk 3]
```diff
@@ -67,128 +24,53 @@ const userController = {
...
+        })
       } else if (user) {
         return res.status(400).json({
-          type: 'Register failed',
-          title: 'Email is used',
-          field_errors: {
-            email: 'used',
-          },
+          status: '400E',
+          message: 'Email is used',
```
Comment (seanlin1125) [inline controllers/user-controller.js:43]: 看要不要改成Email: email is used
或是把它當成資料重複錯誤→400D；Data:email is used

**Review by seanlin1125 — COMMENTED** [review]
你看一下我的3個comment，改一改我再merge
或者是你把剩下的也都寫完我再一起approve?

File: routes/index.js (hunk 1 above)
Comment (AdrieneTZ) [reply routes/index.js:26]: 沒錯，因為驗證資料格式放在 `passport.js` 不太適當，所以拆出來自成一個middleware

File: controllers/user-controller.js (hunk 2 above)
Comment (AdrieneTZ) [reply controllers/user-controller.js:36]: OK，這部分我來改

File: controllers/user-controller.js (hunk 3 above)
Comment (AdrieneTZ) [reply controllers/user-controller.js]: 改成下面這樣：
`status: '400F',
message: 'Field: email has been used.'`

跟其他res json格式統一

**Review by seanlin1125 — APPROVED** [review]
感謝修改~繼續一起努力!

**Pull request merged**